contrib/win32/install/install/components/c_server.wxi

```diff
@@ -0,0 +1,110 @@
+      <util:EventManifest />
+    </File>
+    <util:XmlFile
+      Id="xmlSetResourceFileName"
+      Sequence="1"
+      File="[#opensshevents.man]"
+      Action="setValue"
+      ElementPath="//*[\[]local-name() = 'provider'[\]]"
+      Name="resourceFileName"
+      Value="[#sshagent.exe]"
+      SelectionLanguage="XPath" />
+    <util:XmlFile
+      Id="xmlSetMessageFileName"
+      Sequence="2"
+      File="[#opensshevents.man]"
+      Action="setValue"
+      ElementPath="//*[\[]local-name() = 'provider'[\]]"
+      Name="messageFileName"
+      Value="[#sshagent.exe]"
+      SelectionLanguage="XPath" />
+  </Component>
+  <Component Id="C_SERVER_SSHAGENTSERVICE" Guid="dd45dc2d-cc8a-4eaa-906e-9967e37a3e44">
+    <File Id="sshagent.exe" Source="$(var.SOURCEDIR)/ssh-agent.exe" Name="ssh-agent.exe" KeyPath="yes" />
+    <ServiceInstall
+      Id="SVC_SSHAGENT"
+      Description="Agent to hold private keys used for public key authentication."
+      DisplayName="OpenSSH Authentication Agent"
+      ErrorControl="critical"
+      Interactive="no"
+      Name="ssh-agent"
+      Start="auto"
+      Type="ownProcess"
+      Vital="yes">
+      <util:ServiceConfig
+        ResetPeriodInDays="1"
+        FirstFailureActionType="restart"
+        SecondFailureActionType="restart"
+        ThirdFailureActionType="restart" />
+      <PermissionEx Sddl="D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)(A;;RP;;;AU)" />
+    </ServiceInstall>
+    <ServiceControl
+      Id="SC_SSHAGENT_UNINSTALL"
+      Name="ssh-agent"
+      Stop="uninstall"
+      Remove="uninstall"
+      Wait="no" />
+  </Component>
+  <Component Id="C_SERVER_SSHAGENT_CONTROL" Guid="adebc907-2e8b-4b49-b343-0de71f342a9e" KeyPath="yes">
+    <Condition>
+      <![CDATA[NOT(DONT_START_SSHAGENT OR DONT_START_SERVICES)]]>
+    </Condition>
+    <ServiceControl
+      Id="SC_SSHAGENT_INSTALL"
+      Name="ssh-agent"
+      Start="install"
+      Wait="no" />
+  </Component>
+  <Component Id="C_SERVER_SSHDSERVICE" Guid="6545df79-4f04-4ac3-8c5d-5baa678c94fc">
+    <File Id="sshd.exe" Source="$(var.SOURCEDIR)/sshd.exe" Name="sshd.exe" KeyPath="yes"/>
+    <ServiceInstall
+      Id="SVC_SSHD"
+      Description="SSH protocol based service to provide secure encrypted communications between two untrusted hosts over an insecure network."
+      DisplayName="OpenSSH SSH Server"
+      ErrorControl="critical"
+      Interactive="no"
+      Name="sshd"
+      Start="auto"
+      Type="ownProcess"
+      Vital="yes">
+      <util:ServiceConfig
+        ResetPeriodInDays="1"
+        FirstFailureActionType="restart"
+        SecondFailureActionType="restart"
+        ThirdFailureActionType="restart" />
+    </ServiceInstall>
+    <ServiceControl
+      Id="SC_SSHD_UNINSTALL"
+      Name="sshd"
+      Stop="uninstall"
+      Remove="uninstall"
+      Wait="no" />
+    <fw:FirewallException
```

Comment:
the firewall exception sholud be optional

Reply:
The firewall will only be opened when you install the server component and to me it makes barely any sense to run an OpenSSH server behind a closed firewall (okay, except for local testing). But perhaps this can be moved into it's own separate feature.

Reply:
Is this intended to be relased as a Microsoft product?

Reply:
Would be nice... But I don't believe this gonna happen.

Reply:
Is the server role installed be default?

Reply:
No.. The SSH client programs feature is the default

Reply:
Let's assume that it will eventually be release my Microsoft.  We should minimize the listening ports unless the customer wants the port opened off box.

Reply:
Do you know how the Windows Server SSH Server capability behaves at that point? We should follow it's behavior to be consistent.  If this will be officially released by Microsoft

Reply:
The existing behavior is not my security baseline.